src/renderer/components/screens/WelcomeScreen/UserAgreement.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+
+import ClickableLink from '../../helpers/ClickableLink'
+import useInstantOnboarding from '../../../hooks/useInstantOnboarding'
+import useTranslationFunction from '../../../hooks/useTranslationFunction'
+import {
+  DEFAULT_INSTANCE_PRIVACY_POLICY_URL,
+  isQRWithDefaultInstance,
+} from './chatmailInstances'
+
+import type { QrWithUrl } from '../../../backend/qr'
+
+const URL_REGEX = /(https?:\/\/[^\s]+)/g
+
+function extractLinkFromQrCode(qrWithUrl: QrWithUrl) {
+  if (qrWithUrl.qr.kind !== 'account') {
+    throw new Error('QR needs to be of kind "account"')
+  }
+
+  const matches = qrWithUrl.url.match(URL_REGEX)
+  if (matches && matches.length > 0) {
+    const { origin } = new URL(matches[0])
+    return origin
+  }
```

Comment:
what is this logic for? is `.domain` from core not what we want here?

Reply:
The core unfortunately doesn't give us the URL schema

Reply:
I think https is safe to assume and require here

Reply:
Since this code is already working and fine I wouldn't change it further